WooCommerce/Classes/ViewRelated/Dashboard/Blaze/BlazeCampaignDashboardView.swift

```diff
@@ -65,12 +112,20 @@ struct BlazeCampaignDashboardView: View {
             if case .showProduct(let product) = viewModel.state {
                 ProductInfoView(product: product)
                     .onTapGesture {
-                        productTapped?()
+                        selectedProductID = product.productID
+                        viewModel.shouldShowIntroView = true
                     }
             } else if case .showCampaign(let campaign) = viewModel.state {
                 BlazeCampaignItemView(campaign: campaign, showBudget: false)
                     .onTapGesture {
-                        campaignTapped?()
+                        guard let site = ServiceLocator.stores.sessionManager.defaultSite else {
+                            return
+                        }
+                        let path = String(format: Constants.campaignDetailsURLFormat,
+                                          campaign.campaignID,
+                                          site.url.trimHTTPScheme(),
+                                          BlazeCampaignSource.myStoreSection.rawValue)
+                        selectedCampaignURL = URL(string: path)
```

Comment:
Super nit: Moving this to the view model will let us unit-test the URL. Also, we can inject the site into the view model as suggested in the other comment.

Reply:
Similar to the above comment, this will be handled in a separate PR.